eng/native/tryrun.cmake

```diff
@@ -156,6 +160,15 @@ elseif(TARGET_ARCH_NAME MATCHES "^(armel|arm|armv6|arm64|loongarch64|riscv64|s39
     set_cache_value(HAVE_TERMIOS2_EXITCODE 1)
   elseif (TIZEN)
     set_cache_value(HAVE_FUNCTIONAL_PTHREAD_ROBUST_MUTEXES_EXITCODE 0)
+  elseif(HAIKU)
+    set_cache_value(GETPWUID_R_SETS_ERRNO_EXITCODE 1)
+    set_cache_value(HAVE_CLOCK_MONOTONIC_COARSE_EXITCODE 1)
+    set_cache_value(HAVE_COMPATIBLE_EXP_EXITCODE 0)
+    set_cache_value(HAVE_COMPATIBLE_ILOGBNAN_EXITCODE 0)
+    set_cache_value(HAVE_PROCFS_MAPS_EXITCODE 1)
+    set_cache_value(HAVE_PROCFS_STAT_EXITCODE 1)
+    set_cache_value(HAVE_PROCFS_STATUS_EXITCODE 1)
+    set_cache_value(SSCANF_CANNOT_HANDLE_MISSING_EXPONENT_EXITCODE 1)
```

Comment:
`SSCANF_CANNOT_HANDLE_MISSING_EXPONENT` and some others of these config values are no longer used by the repo. 

Bonus points for cleaning them up from everywhere.

Reply:
I see that the checks are still available in the `configure.cmake` files, such as [here](https://github.com/dotnet/runtime/blob/d83506d6880a8a420ccf7b951bd8c7e430251734/src/coreclr/pal/src/configure.cmake#LL212C1-L226C42). Should they be removed as well?

Reply:
Along with `SSCANF_CANNOT_HANDLE_MISSING_EXPONENT`, I can see that:
- `GETPWUID_R_SETS_ERRNO_EXITCODE`
- `HAVE_PROCFS_MAPS`
- `HAVE_PROCFS_STATUS`
also do not appear anywhere else other than `configure.cmake`.

Reply:
> Bonus points for cleaning them up from everywhere.

I've removed all of the above values from `eng/native/tryrun**.cmake`. It does not break compilation for Haiku, hopefully not for all the other platforms.